L/Lua/build_tarballs.jl

```diff
@@ -0,0 +1,47 @@
+using BinaryBuilder
+
+name = "Lua"
+version = v"5.3.5"
+
+sources = [
+    "https://www.lua.org/ftp/lua-5.3.5.tar.gz" =>
+        "0c2eed3f960446e1a3e4b9a1ca2f3ff893b6ce41942cf54d5dd59ab4b3b058ac",
+]
+
+script = raw"""
+cd ${WORKSPACE}/srcdir/lua-*
+
+if [[ ${target} == *-apple-* ]]; then
+    MAKE_TARGET=macosx
+elif [[ ${target} == *-freebsd* ]]; then
+    # NOTE: The FreeBSD ports build also uses the bsd instead of the freebsd target,
+    # since it hard-codes fewer things, apparently
+    MAKE_TARGET=bsd
+elif [[ ${target} == *-mingw* ]]; then
+    MAKE_TARGET=mingw
+else
+    MAKE_TARGET=linux
+fi
+
+# XXX: Work around Lua apparently not caring that Windows uses .exe
+if [[ ${target} == *-mingw* ]]; then
+    TO_BIN="lua.exe luac.exe"
+else
+    TO_BIN="lua luac"
+fi
+
+make -j${nproc} ${MAKE_TARGET}
+make install INSTALL_TOP="${prefix}" INSTALL_LIB="${libdir}" TO_BIN="${TO_BIN}"
```

Comment:
```suggestion
make install INSTALL_TOP="${prefix}" INSTALL_LIB="${libdir}" TO_BIN="${TO_BIN}" TO_LIB="${TO_LIB}"
```
and you need to define
```
TO_LIB=lua53.${dlext}
```
for Windows and
```
TO_LIB=liblua.${ldext}
```
for the rest of the world (at least this is what I expect)

Reply:
That should work if I patch it to build the shared library instead of the archive. If I add the shared library alongside the archive, I think it would need to be
```sh
TO_LIB="liblua.a liblua.${dlext}"
```
for the non-Windows case.

Reply:
Do we need to install the archive?  Usually we ignore it if possible

Reply:
No I don't think we need to install it, but we need to at least build it, unless I do what Homebrew does and patch it to dynamically link liblua to the Lua executable rather than statically link it.

Reply:
Oh, are the `TO_*` variables only used for installation?

Reply:
Haaaaaa they are

Reply:
> but we need to at least build it

Fine, but this is for `make install` :slightly_smiling_face: